web/src/pages/sample/SampleView.tsx

```diff
@@ -0,0 +1,105 @@
+import Pedigree from '../../shared/components/pedigree/Pedigree'
+import SeqPanel from '../../shared/components/SeqPanel'
+import MuckError from '../../shared/components/MuckError'
+import SampleInfo from '../../shared/components/SampleInfo'
+
+const GET_SAMPLE_INFO = gql(`
+query SampleInfo($sample_id: String!) {
+    sample(id: $sample_id) {
+      id
+      participantId
+      externalId
+      participant {
+        externalId
+        families {
+            id
+        }
+      }
+      sequences {
+        id
+        externalIds
+        type
+        meta
+      }
+      type
+      active
+    }
+  }
+`)
+
+const sampleFieldsToDisplay = ['active', 'type', 'participantId']
+
+const SampleView: React.FunctionComponent<Record<string, unknown>> = () => {
+    const { sampleName } = useParams()
+    const sampleID = sampleName || ''
+
+    const { loading, error, data } = useQuery(GET_SAMPLE_INFO, {
+        variables: { sample_id: sampleID },
+    })
+
+    if (loading) return <LoadingDucks />
+    if (error) return <>Error! {error.message}</>
+
+    return data ? (
+        <>
+            <div className="dataStyle">
+                <div
+                    style={{
+                        borderBottom: `1px solid black`,
+                    }}
+                >
+                    <h1
+                        style={{
+                            display: 'inline',
+                        }}
+                        key={`${data.sample.participant?.externalId}`}
+                    >
+                        {`${data.sample.participant?.externalId}\t`}
+                    </h1>
+                    <h3
+                        style={{
+                            display: 'inline',
+                        }}
+                    >
+                        {`${data.sample.id}\t${data.sample.externalId}`}
+                    </h3>
+                </div>
+                <div style={{ paddingBottom: '20px' }}>
+                    {data?.sample.participant?.families.map((family) => (
+                        <React.Fragment key={family.id}>
+                            <Pedigree familyID={family.id} />
+                        </React.Fragment>
+                    ))}
+                </div>
+                <div style={{ paddingBottom: '20px' }}>
+                    <h4
+                        style={{
+                            borderBottom: `1px solid black`,
+                        }}
+                    >
+                        Sample Information
+                    </h4>
+                    <SampleInfo
+                        sample={Object.fromEntries(
+                            Object.entries(data.sample).filter(([key]) =>
+                                sampleFieldsToDisplay.includes(key)
+                            )
+                        )}
+                    />
+                </div>
+                <SeqPanel isOpen sequences={data.sample.sequences} />
```

Comment:
Just curious - is `data.sample.sequences` guaranteed by the API to always be present? Might cause access errors if not. Also applies for other areas in the code with chained property access.

Reply:
Yeah it should always be present, could be an empty array though. These are all validated by the typescript graphql codegen